### PR TITLE
Support destroy plans in temporal workflow.

### PR DESCRIPTION
**Pull request by nishkrishnan**
Creates a new field called planMode which pertains to this spec:
https://www.terraform.io/cli/commands/plan#planning-modes

### CODE REVIEW

File: server/neptune/workflows/internal/deploy/workflow.go [hunk 1]
```diff
@@ -88,8 +88,8 @@ func newRunner(ctx workflow.Context, request Request, tfWorkflow terraform.Workf
 	// can potentially change a root configuration
 	root := root.Root{
 		Name:      request.Root.Name,
-		Apply:     job.Job{Steps: convertToInternalSteps(request.Root.Apply.Steps)},
```
Comment (Aayyush) [inline server/neptune/workflows/internal/deploy/workflow.go:91]: nit: can we add new lines to make this more readable?

File: server/neptune/workflows/internal/root/root.go [hunk 2]
```diff
@@ -12,8 +12,8 @@ type Root struct {
 	// Path is the relative path from the repo
 	Path      string
 	TfVersion string
-	Apply     job.Job
```
Comment (Aayyush) [inline server/neptune/workflows/internal/root/root.go:15]: thoughts on wrapping this in a `job.Apply` struct for readability?

**Review by Aayyush — APPROVED** [review]
left minor nits

File: server/neptune/workflows/internal/deploy/workflow.go (hunk 1 above)
Comment (nishkrishnan) [reply server/neptune/workflows/internal/deploy/workflow.go:91]: sure can do that.

File: server/neptune/workflows/internal/root/root.go (hunk 2 above)
Comment (nishkrishnan) [reply server/neptune/workflows/internal/root/root.go:15]: not sure how that really helps with readability?

Comment (samrabelachew) [reply server/neptune/workflows/internal/root/root.go:15]: maybe keep `job.Terraform` for the apply job/general job and rename the plan specific struct to`job.TerraformPlan` (assuming you mean trying to maintain the connection between the two job structs)? either way I think it's okay

**Review by samrabelachew — APPROVED** [review]

**Pull request merged**